sdk/nodejs/resource.ts

```diff
@@ -703,6 +703,51 @@ export abstract class ProviderResource extends CustomResource {
     }
 }
 
+/**
+ * A reference to a `ProviderResource`.  Should be used as much as possible when referring to
+ * `Provider`s from other locations.  For example, when specifying the `Provider` for a `Resource`
+ * or the `Provider` for an `data source` `invoke` call, a `ProviderRef` should be provided instead
+ * of a `Provider`.
+ *
+ * A `ProviderRef` can be obtained by calling `await ProviderRef.get(provider)`.
+ */
+export class ProviderRef {
+    // tslint:disable-next-line: variable-name
+    public readonly __providerRefInstance = true;
+
+    /** @internal */
+    public getPackage: () => string;
+
+    /** @internal */
+    public getValue: () => string;
+
+    /**
+     * Asynchronously creates a new `ProviderRef` for the given `ProviderResource`.
+     */
+    public static async get(provider: ProviderResource): Promise<ProviderRef> {
+        const providerURN = await provider.urn.promise();
+        const providerID = await provider.id.promise() || unknownValue;
+        return new ProviderRef(provider.getPackage(), `${providerURN}::${providerID}`);
+    }
```

Comment:
note: i implemented it this way (as opposed to an instance method on `Provider`) so that this will work in sxs scenarios.  Code in this layer that needs a ProviderRef can always obtain one from a provider, no matter if that provider subclasses `Provider` from this package, or from a sxs package.